homeassistant/components/todoist/config_flow.py

```diff
@@ -0,0 +1,63 @@
+"""Config flow for todoist integration."""
+
+from http import HTTPStatus
+import logging
+from typing import Any
+
+from requests.exceptions import HTTPError
+from todoist_api_python.api_async import TodoistAPIAsync
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_TOKEN
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SETTINGS_URL = "https://todoist.com/app/settings/integrations"
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_TOKEN): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for todoist."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            api = TodoistAPIAsync(user_input[CONF_TOKEN])
+            try:
+                await api.get_tasks()
+            except HTTPError as err:
+                if err.response.status_code == HTTPStatus.UNAUTHORIZED:
+                    errors["base"] = "invalid_access_token"
+                else:
+                    errors["base"] = "cannot_connect"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                await self.async_set_unique_id(user_input[CONF_TOKEN])
```

Comment:
Why do we set a unique id if we only allow one config entry? And the inverse question, why only allow one config entry if we can set a unique id?

Reply:
Second question first: I first set this up with a unique id then saw the service call adds tasks without referencing a config entry or entity, and so that looked like a problem.
(The config entry however does reference a project name, so it could be possible perhaps to infer but that seemed sketchy).

First question: However, I thought it might be possible to add more config entries in the future if the service changes or is deprecated so I left the unique id. I didn't think it hurts, is there a down side?

As we talk through, I also realize this could potentially be updated in the future to support re-auth updating the API key so maybe this isn't a valid in unique id.  I was thinking of it like an oauth client id but realize how it's not really like than. Given that, perhaps needs to be removed anyway. I'm not sure what, then, could be a unique id but theoretically I'd like this to be able to support multiple accounts since once the service is fixed there's no big reason it couldn't otherwise.